### PR TITLE
Fix kind warnings

### PR DESCRIPTION
**Pull request by Thimoteus**

### CODE REVIEW

**Comment by hdgarrood** [comment]
These are fixed in the 0.14 branch aren’t they?

**Comment by JordanMartinez** [comment]
@hdgarrood No, not yet. My previous PR came **before** the changes Nate made when core contributors decided to emit the "could not generalize kind" warnings.

File: src/Data/NaturalTransformation.purs [hunk 1]
```diff
@@ -14,6 +14,7 @@ module Data.NaturalTransformation where
 -- | `f` and `g` should be functors, but the `Functor` constraint is not
 -- | enforced here; that the types are of kind `Type -> Type` is enough for our
 -- | purposes.
+type NaturalTransformation :: forall k. (k -> Type) -> (k -> Type) -> Type
```
Comment (hdgarrood) [inline src/Data/NaturalTransformation.purs:17]: This kind signature disagrees with the doc-comments. I feel like perhaps we should make this `(Type -> Type) -> (Type -> Type) -> Type` unless there is a specific example where the generalized version would be useful.

Comment (hdgarrood) [reply src/Data/NaturalTransformation.purs:17]: (Or alternatively update the doc comments)

Comment (Thimoteus) [reply src/Data/NaturalTransformation.purs:17]: That's a good point, I've updated the doc comments.

**Comment by hdgarrood** [comment]
I don’t remember closing this; I guess must have closed it by accident, sorry!

**Pull request closed without merging**